frontend/src/components/chats/components/current-chat/components/messages-list/messages-list.tsx

```diff
@@ -11,21 +11,49 @@ type Props = {
 };
 
 const MessagesList: FC<Props> = ({ messages, currentUserId }) => {
+  const msgs: JSX.Element[] = [];
+  const filteredMsgs: JSX.Element[] = [];
+
+  messages.map((message) => {
```

Comment:
Won't it be easier to group messages inside reduce by a date and then show them with separator by that grouping key?
After grouping you'd have a structure like this `DateToMessagesMap = { [<msg.createdAt>(grouping_key)]: <message_obj>[] }`
reduce would look something like that
```
const groupedByDateMessages = messages.reduce<DateToMessagesMap>((dateToMessages, message) => {
  const formattedMessageDate = getFormattedDate(message.createdAt, 'yyyy-MM-dd');
  return {
    ...dateToMessages,
    [formattedMessageDate]: (dateToMessages?.[formattedMessageDate] ?? []).concat(message)
  };
}, {});
```
and to extract grouped by date messages you'd have something like that
```
const msgsToDisplay = Object.entries(groupedByDateMessages).map((groupingObject) => {
  const [groupingDate, groupedMessages] = groupingObject;
  
  return (
    <>
      <DateSeparator postTime={groupingDate} />
      {groupedMessages.map(message => {
        return <Message {...message_props_stuff_from_message_obj}>
      })}
    </>
  );
});
```

Reply:
this approach is awesome, thank you, Andrew!